### ISSUE TITLE
Replicating RobertaForSequenceClassification model

### ISSUE THREAD

**Issue by ganeshjawahar** [nyu-mll/jiant#1109]
Hello,

Thanks for the wonderful tool. I just started using it for my research. I am trying to replicate HF's RobertaForSequenceClassification (RSC) model in jiant, but I observe large difference in classification performance. HF's RSC accuracy is 93.3 while jiant's is 88.3. I tried to match all the hyperparameters, play with MLP module a bit to match. Any suggestions?

I am providing as much relevant info. as possible, but pls. feel free to ask me if you need something more.

In constructor of classifier class at jiant/modules/simple_modules.py I defined new classifier as RSC doesn't use nn.Tanh and nn.LayerNorm:
`cls_type == "det_mlp":
            classifier = nn.Sequential(
                nn.Linear(d_inp, d_hid),
                nn.Dropout(dropout),
                nn.Linear(d_hid, n_classes),
            )`

My config file is:
include "defaults.conf"
exp_name = stilts-demo
run_name = first-run
pretrain_tasks = "classifier"
target_tasks = "classifier"

//Experiment configs
do_pretrain = 1
do_target_task_training = 0
do_full_eval = 1

batch_size = 24
max_seq_len = 128

write_preds = "val,test"

//RoBERTa-specific parameters
input_module = "roberta-large"
transformers_output_mode = "top"
sep_embs_for_skip = 1
sent_enc = none
classifier = det_mlp
classifier_hid_dim = 1024
classifier_dropout = 0.1

dropout = 0.1 // following BERT paper
optimizer = bert_adam
lr_decay_factor = 0 // my reference implementation doesn't do decay
max_epochs = 3
lr = .00001
min_lr = .0000001
lr_patience = 4
patience = 20
max_vals = 10000
transfer_paradigm = "finetune"

jiant's RoBERTa large model contain _highway_layer and _phrase_layer at the end while RSC RoBERTa large model doesn't. I don't know if I need to bridge the gap:

jiant's model (last few layers):
)
              (23): BertLayer(
                (attention): BertAttention(
                  (self): BertSelfAttention(
                    (query): Linear(in_features=1024, out_features=1024, bias=True)
                    (key): Linear(in_features=1024, out_features=1024, bias=True)
                    (value): Linear(in_features=1024, out_features=1024, bias=True)
                    (dropout): Dropout(p=0.1)
                  )
                  (output): BertSelfOutput(
                    (dense): Linear(in_features=1024, out_features=1024, bias=True)
                    (LayerNorm): LayerNorm(torch.Size([1024]), eps=1e-05, elementwise_affine=True)
                    (dropout): Dropout(p=0.1)
                  )
                )
                (intermediate): BertIntermediate(
                  (dense): Linear(in_features=1024, out_features=4096, bias=True)
                )
                (output): BertOutput(
                  (dense): Linear(in_features=4096, out_features=1024, bias=True)
                  (LayerNorm): LayerNorm(torch.Size([1024]), eps=1e-05, elementwise_affine=True)
                  (dropout): Dropout(p=0.1)
                )
              )
            )
          )
          (pooler): BertPooler(
            (dense): Linear(in_features=1024, out_features=1024, bias=True)
            (activation): Tanh()
          )
        )
      )
      **(_highway_layer): TimeDistributed(
        (_module): Highway(
          (_layers): ModuleList()
        )
      )
      (_phrase_layer): NullPhraseLayer()
      (_dropout): Dropout(p=0.1)**
    )
    (classifier_mdl): SingleClassifier(
      (pooler): Pooler()
      (classifier): Classifier(
        (classifier): Sequential(
          (0): Linear(in_features=1024, out_features=1024, bias=True)
          (1): Dropout(p=0.1)
          (2): Linear(in_features=1024, out_features=2, bias=True)
        )
      )
    )
  )
)

HF's RSC last few layers:
(23): BertLayer(
          (attention): BertAttention(
            (self): BertSelfAttention(
              (query): Linear(in_features=1024, out_features=1024, bias=True)
              (key): Linear(in_features=1024, out_features=1024, bias=True)
              (value): Linear(in_features=1024, out_features=1024, bias=True)
              (dropout): Dropout(p=0.1, inplace=False)
            )
            (output): BertSelfOutput(
              (dense): Linear(in_features=1024, out_features=1024, bias=True)
              (LayerNorm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
              (dropout): Dropout(p=0.1, inplace=False)
            )
          )
          (intermediate): BertIntermediate(
            (dense): Linear(in_features=1024, out_features=4096, bias=True)
          )
          (output): BertOutput(
            (dense): Linear(in_features=4096, out_features=1024, bias=True)
            (LayerNorm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
            (dropout): Dropout(p=0.1, inplace=False)
          )
        )
      )
    )
    (pooler): BertPooler(
      (dense): Linear(in_features=1024, out_features=1024, bias=True)
      (activation): Tanh()
    )
  )
  (classifier): RobertaClassificationHead(
    (dense): Linear(in_features=1024, out_features=1024, bias=True)
    (dropout): Dropout(p=0.1, inplace=False)
    (out_proj): Linear(in_features=1024, out_features=2, bias=True)
  )
)

Thanks in advance. 

Appreciate your time.

**Comment by sleepinyourhat** [comment]
Hrm—it'll be easiest for us to help if you can show an issue with a supported task. We're focusing development on v2 (jiant.info), so we don't have that much bandwidth for in-depth debugging, sadly.

It does look like you're getting an extra dropout, though I'd be a bit surprised if that would hurt performance as much as you're reporting. Do take a close look at the optimizer setup as well.

@HaokunLiu, if you recognize anything obviously worrying, do chime in.
